modules/deepintentDpesIdSystem.js

```diff
@@ -0,0 +1,80 @@
+/**
+ * This module adds DPES to the User ID module
+ * The {@link module:modules/userId} module is required
+ * @module modules/deepintentDpesSystem
+ * @requires module:modules/userId
+ */
+
+import { submodule } from '../src/hook.js';
+import { getStorageManager } from '../src/storageManager.js';
+
+const MODULE_NAME = 'deepintentId';
+export const storage = getStorageManager(null, MODULE_NAME);
+const DEEPINTENT_DPES_ID = 'di_dpes';
+
+function readCookie(cookieName) {
+  const val = storage.cookiesAreEnabled ? storage.getCookie(cookieName) : null;
+  return JSON.parse(val);
+}
+
+function readFromLocalStorage() {
+  const val = storage.localStorageIsEnabled
+    ? storage.getDataFromLocalStorage(DEEPINTENT_DPES_ID)
+    : null;
+  return JSON.parse(val);
+}
+
+/** @type {Submodule} */
+export const deepintentDpesSubmodule = {
+  /**
+   * used to link submodule with config
+   * @type {string}
+   */
+  name: MODULE_NAME,
+  /**
+   * decode the stored id value for passing to bid requests
+   * @function
+   * @param {{value:string}} value
+   * @returns {{deepintentId:Object}}
+   */
+  decode(value, config) {
+    const configParams = (config && config.params) || {};
+    if (configParams && configParams.identityKey && configParams.siteId && value.siteId && configParams.siteId == value.siteId) {
+      if (configParams.identityKey === 'hashedEmail') {
+        value = {'deepintentId': value.email}
+        return value;
+      } else if (configParams.identityKey === 'hashedNPI') {
+        value = {'deepintentId': value.npi}
+        return value;
+      } else {
+        return undefined;
+      }
+    } else {
+      return undefined;
+    }
+  },
+  /**
+   * performs action to obtain id and return a value in the callback's response argument
+   * @function
+   * @param {SubmoduleConfig} config
+   * @return {{id: string | undefined} | undefined}
+   */
+  getId(config) {
+    const configStorage = (config && config.storage) || {};
+
+    let id = null;
+    if (
+      configStorage &&
+      configStorage.name &&
+      configStorage.type === 'cookie'
+    ) {
+      id = readCookie(configStorage.name);
```

Comment:
why are you manually reading from the cookie if you are asking publishers to set the `storage` config?  the userId module will already handle storage and retrieval you just need to add the `cacheIdObj` to the `getId()` signature

Reply:
This code was started when cacheIdObj signature was not there, the code will be later changed to accomodate this in phase 2.

Reply:
As a total change I will be changing the description in Doc and adding unit test cases in eids_spec.js and userId_spec.js thanks for letting me know.

Reply:
the `storedId` has been part of the signature for years now...

Reply:
Why wait for a phase two,  I'm in the do it right the first time camp

Reply:
storage will not be handled by UserModule as we do not use any http calls to be made to retrieve id from server and store it via user module. This id is stored by our select publishers side which needs to register on our platform and thus we have to read from storage manually, thus keeping the structure same as per the plan.

Reply:
changes are completed as suggested.